core/domain/user_query_jobs_one_off_test.py

```diff
@@ -0,0 +1,157 @@
+            'property_name': 'objective',
+            'new_value': 'the objective'
+        }], 'Test edit')
+
+        self.save_new_valid_exploration(
+            self.EXP_ID_2, self.user_d_id, end_state_name='End')
+
+        exp_services.update_exploration(self.user_d_id, self.EXP_ID_2, [{
+            'cmd': 'edit_exploration_property',
+            'property_name': 'objective',
+            'new_value': 'the objective'
+        }], 'Test edit')
+
+        # set tmpsuperadm1n as admin in ADMIN_USERNAMES config property.
+        self.set_admins(['tmpsuperadm1n'])
+
+    def test_user_is_active_in_last_n_days(self):
+        query_id = user_query_services.save_new_query_model(
+            self.submitter_id, active_in_last_n_days=3)
+        self._run_one_off_job(query_id)
+
+        query = user_models.UserQueryModel.get(query_id)
+        qualifying_user_ids = [self.user_b_id, self.user_c_id, self.user_d_id]
+        self.assertEqual(len(query.user_ids), 3)
+        self.assertEqual(sorted(query.user_ids), sorted(qualifying_user_ids))
+
+    def test_user_has_created_at_least_n_exps(self):
+        query_id = user_query_services.save_new_query_model(
+            self.submitter_id, created_at_least_n_exps=1)
+        self._run_one_off_job(query_id)
+
+        query = user_models.UserQueryModel.get(query_id)
+        qualifying_user_ids = [self.user_b_id, self.user_d_id]
+        self.assertEqual(len(query.user_ids), 2)
+        self.assertEqual(sorted(query.user_ids), sorted(qualifying_user_ids))
+
+    def test_user_has_created_fewer_than_n_exps(self):
+        query_id = user_query_services.save_new_query_model(
+            self.submitter_id, created_fewer_than_n_exps=1)
+        self._run_one_off_job(query_id)
+
+        query = user_models.UserQueryModel.get(query_id)
+        qualifying_user_ids = [self.user_a_id, self.user_c_id]
+        self.assertEqual(len(query.user_ids), 2)
+        self.assertEqual(sorted(query.user_ids), sorted(qualifying_user_ids))
+
+    def test_user_has_edited_at_least_n_exps(self):
+        query_id = user_query_services.save_new_query_model(
+            self.submitter_id, edited_at_least_n_exps=1)
+        self._run_one_off_job(query_id)
+
+        query = user_models.UserQueryModel.get(query_id)
+        qualifying_user_ids = [self.user_b_id, self.user_c_id, self.user_d_id]
+        self.assertEqual(len(query.user_ids), 3)
+        self.assertEqual(sorted(query.user_ids), sorted(qualifying_user_ids))
+
+    def test_user_has_edited_fewer_than_n_exps(self):
+        query_id = user_query_services.save_new_query_model(
+            self.submitter_id, edited_fewer_than_n_exps=1)
+        self._run_one_off_job(query_id)
+
+        query = user_models.UserQueryModel.get(query_id)
+        qualifying_user_ids = [self.user_a_id]
+        self.assertEqual(len(query.user_ids), 1)
+        self.assertEqual(query.user_ids, qualifying_user_ids)
+
+    def test_combination_of_query_params(self):
```

Comment:
To test this properly, can you find a way to set this up so that there are two query options A and B, and the result is not the same result as you'd get from executing either A or B in isolation?